src/main/java/com/ita/if103java/ims/dao/impl/ItemDaoImpl.java

```diff
@@ -26,9 +28,11 @@ public ItemDaoImpl(DataSource dataSource, ItemRowMapper itemRowMapper) {
     }
 
     @Override
-    public List<Item> getItems() {
+    public List<Item> getItems(Pageable pageable) {
         try {
-            return jdbcTemplate.query(Queries.SQL_SELECT_ITEMS, itemRowMapper);
+            String sort = pageable.getSort().toString().replaceAll(": ", " ");
+            return jdbcTemplate.query(Queries.SQL_SELECT_ITEMS, itemRowMapper, sort, pageable.getPageSize(),
+                pageable.getPageNumber());
```

Comment:
need to multiply pageSize and pageNumber, because offset is how many rows to skip and not how many pages to skip

Reply:
or use pageable.getOffset()